scripts/extract_download_logs.py

```diff
@@ -57,7 +80,32 @@
     "rp_start",
     "rp_end",
 ]
-OUTPUT_FILENAME = options.filename
+OUTPUT_FILENAME = args.filename
+
+
+def send_notify(
+    from_date: datetime.datetime,
+    to_date: datetime.datetime,
+    file: StringIO,
+    api_key: str = os.environ["NOTIFY_API_KEY"],
```

Comment:
If `NOTIFY_API_KEY` missing from environment, this causes the whole script to fail immediately. This forces users to have that key present even if they are not interested in the email report option. Given that the email report is optional, can we modify the logic so that users are able to run the script without the Notify API key? We could then perhaps add some guidance to the `email` arg to mention that the Notify API key is necessary for this to work, and handle the situation more gracefully when the `email` arg is selected but no Notify API key is present. Ideally we would also output some instructions on how to obtain this API key. From the [Notify documentation](https://www.notifications.service.gov.uk/using-notify/api-documentation) I can see the following instructions:

To create an API key:

1. [Sign in to GOV.UK Notify](https://www.notifications.service.gov.uk/sign-in).
2. Go to the API integration page.
3. Select API keys.
4. Select Create an API key.

Reply:
Good spot, have added the soft lookup in: 1823a84